src/soroban/assembled_transaction.ts

```diff
@@ -241,71 +304,55 @@ export class AssembledTransaction<T> {
     return new AssembledTransaction.Result.Err(err);
   }
 
-  getPublicKey = async (): Promise<string | undefined> => {
-    const wallet = this.options.wallet;
-    if (!(await wallet.isConnected()) || !(await wallet.isAllowed())) {
-      return undefined;
-    }
-    return (await wallet.getUserInfo()).publicKey;
-  };
-
   /**
-   * Get account details from the Soroban network for the publicKey currently
-   * selected in user's wallet. If not connected to Freighter, use placeholder
-   * null account.
-   */
-  getAccount = async (): Promise<Account> => {
-    const publicKey = await this.getPublicKey();
-    return publicKey
-      ? await this.server.getAccount(publicKey)
-      : new Account(NULL_ACCOUNT, "0");
-  };
-
-  /**
-   * Sign the transaction with the `wallet` (default Freighter), then send to
-   * the network and return a `SentTransaction` that keeps track of all the
-   * attempts to send and fetch the transaction from the network.
+   * Sign the transaction with the `wallet`, included previously. If you did
+   * not previously include one, you need to include one now that at least
+   * includes the `signTransaction` method. After signing, this method will
+   * send the transaction to the network and return a `SentTransaction` that
+   * keeps track of all the attempts to fetch the transaction.
    */
   signAndSend = async ({
-    secondsToWait = 10,
     force = false,
+    wallet = this.options.wallet,
   }: {
-    /**
-     * Wait `secondsToWait` seconds (default: 10) for both the transaction to SEND successfully (will keep trying if the server returns `TRY_AGAIN_LATER`), as well as for the transaction to COMPLETE (will keep checking if the server returns `PENDING`).
-     */
-    secondsToWait?: number;
     /**
      * If `true`, sign and send the transaction even if it is a read call.
      */
     force?: boolean;
+    /**
+     * The wallet to use for signing. If not provided, the wallet from the
+     * options will be used. You must provide a wallet here if you did not
+     * provide one before, but the only method it needs to include at this
+     * point is `signTransaction`.
+     */
+    wallet?: Pick<Wallet, "signTransaction">;
   } = {}): Promise<SentTransaction<T>> => {
-    if (!this.raw) {
+    if (!this.built) {
       throw new Error("Transaction has not yet been simulated");
     }
 
     if (!force && this.isReadCall) {
-      throw new Error(
+      throw new AssembledTransaction.Errors.NoSignatureNeeded(
```

Comment:
I don't get it - can't you just skip the sign step?

Reply:
Yes, this is for if people call `signAndSend` on a read-only transaction. They have the option of forcing inclusion in the ledger with `force: true`